src/app/screens/gamepage/index.ts

```diff
@@ -178,6 +178,20 @@ export class GamePage {
       })
     })
   }
+  private recruitUnits() {
+    this.unitsButton = document.querySelector(
+      '.units-button'
+    ) as HTMLButtonElement
+    if (this.game.possibleToAddUnit()) {
+      this.unitsButton.addEventListener('click', () => {
+        this.game.addVillageUnit(new Knight())
+      })
+    }
+    else {
+      this.unitsButton.style.visibility = 'hidden';
```

Comment:
button should be hidden before clicking it so we need to place method to hide or show inside method that is responsible for updating ui after attack @bartoszbialecki